packages/service/src/doc-updates.test.ts

```diff
@@ -0,0 +1,35 @@
+import * as Y from "yjs";
+
+test("persist equivalent final crdt via different update paths", async () => {
+  // create a yjs document A
+  const ydoc1 = new Y.Doc();
+  const ydoc2 = new Y.Doc();
+
+  // make a change to the document
+  const ytext1 = ydoc1.getText("codemirror");
+  const ytext2 = ydoc2.getText("codemirror");
+  ytext1.insert(0, "foo bar");
+  ytext2.insert(0, "hello world");
+
+  // encode the current state as a binary buffer and destory docs
+  const currentState1 = Y.encodeStateAsUpdate(ydoc1);
+  const currentState2 = Y.encodeStateAsUpdate(ydoc2);
+  ydoc1.destroy();
+  ydoc2.destroy();
+
+  // calculate diffs
+  const stateVector1 = Y.encodeStateVectorFromUpdate(currentState1);
+  const stateVector2 = Y.encodeStateVectorFromUpdate(currentState2);
+  const diff1 = Y.diffUpdate(currentState1, stateVector2);
+  const diff2 = Y.diffUpdate(currentState2, stateVector1);
```

Comment:
I'm confused about what this is testing. I think the diffs we care about are the diffs between the original state and the updated state. 

diff1: the diff between state1 origin and state1 updated
diff2: the diff between state2 origin and state2 updated

expect that applying diff2 to state1 will produce the same result as applying diff1 to state2.

Reply:
The plan is to store these diffs in the `document_updates_queue` table? Let's write a test for that also.